Gems/Atom/RHI/Code/Source/RHI/PipelineState.cpp

```diff
@@ -50,8 +63,10 @@ namespace AZ::RHI
             [this, &descriptor, &pipelineLibrary, &resultCode](int deviceIndex)
             {
                 auto* device = RHISystemInterface::Get()->GetDevice(deviceIndex);
```

Comment:
can be const + I believe auto is only mostly used for iterators in this codebase